Mage.Tests/src/test/java/org/mage/test/cards/single/who/ThijarianWitnessTest.java

```diff
@@ -0,0 +1,171 @@
+        addCard(Zone.BATTLEFIELD, playerA, tiny2);
+        addCard(Zone.BATTLEFIELD, playerB, "Badlands", 4);
+        addCard(Zone.HAND, playerB, kill, 2);
+
+        attack(1, playerA, tiny);
+        attack(1, playerA, tiny2);
+        castSpell(1, PhaseStep.DECLARE_ATTACKERS, playerB, kill, tiny, true);
+        castSpell(1, PhaseStep.DECLARE_ATTACKERS, playerB, kill, tiny2, true);
+
+        setStopAt(1, PhaseStep.END_TURN);
+        execute();
+
+        assertPermanentCount(playerA, clue, 1);
+        assertExileCount(playerA, 1);
+    }
+    @Test
+    public void test_BlockingAloneAfterKill() {
+        setStrictChooseMode(true);
+        addCard(Zone.BATTLEFIELD, playerA, witness);
+        addCard(Zone.BATTLEFIELD, playerA, big);
+        addCard(Zone.BATTLEFIELD, playerA, "Badlands", 4);
+        addCard(Zone.HAND, playerA, kill);
+        addCard(Zone.BATTLEFIELD, playerB, tiny);
+        addCard(Zone.BATTLEFIELD, playerB, tiny2);
+
+        attack(1, playerA, big);
+        block(1, playerB, tiny, big);
+        block(1, playerB, tiny2, big);
+        castSpell(1, PhaseStep.DECLARE_BLOCKERS, playerA, kill, tiny2);
+
+        setStopAt(1, PhaseStep.END_TURN);
+        execute();
+
+        assertPermanentCount(playerA, clue, 1);
+        assertPermanentCount(playerB, tiny, 0);
+        assertPermanentCount(playerA, big, 1);
+        assertExileCount(playerB, 1);
+    }
+    @Test
+    public void test_DoubleBlocked() {
+        //Auto-assign damage
+        //setStrictChooseMode(true);
+        addCard(Zone.BATTLEFIELD, playerA, witness);
+        addCard(Zone.BATTLEFIELD, playerA, big);
+        addCard(Zone.BATTLEFIELD, playerB, tiny);
+        addCard(Zone.BATTLEFIELD, playerB, tiny2);
+
+        attack(1, playerA, big);
+        block(1, playerB, tiny, big);
+        block(1, playerB, tiny2, big);
+
+        setStopAt(1, PhaseStep.END_TURN);
+        execute();
+
+        assertPermanentCount(playerA, clue, 1);
+        assertPermanentCount(playerB, tiny, 0);
+        assertPermanentCount(playerA, big, 0);
+        assertExileCount(playerA, 1);
+        assertExileCount(playerB, 0);
+    }
+    @Test
+    public void test_DoubleBlocker() {
+        //Auto-assign damage
+        //setStrictChooseMode(true);
+        addCard(Zone.BATTLEFIELD, playerA, witness);
+        addCard(Zone.BATTLEFIELD, playerA, tiny);
+        addCard(Zone.BATTLEFIELD, playerA, tiny2);
+        addCard(Zone.BATTLEFIELD, playerB, "Night Market Guard");
+
+        attack(1, playerA, tiny);
+        attack(1, playerA, tiny2);
+        block(1, playerB, "Night Market Guard", tiny);
+        block(1, playerB, "Night Market Guard", tiny2);
+
+        setStopAt(1, PhaseStep.END_TURN);
+        execute();
+
+        assertPermanentCount(playerA, clue, 1);
+        assertExileCount(playerA, 0);
+        assertExileCount(playerB, 1);
+    }
+    @Test
+    public void test_AttackAndBlock() {
+        //auto-stack triggers
```

Comment:
setChoice(player, string) where string is the trigger text should work for stacking triggers?

Reply:
Good point. I assume it's not a problem to do `setChoice(playerA, "");`?